xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1025,6 +1067,12 @@ bool DX::DeviceResources::DoesTextureSharingWork()
     CSysInfo::GetWindowsDeviceFamily() != CSysInfo::Desktop)
     return false;
 
+  DXGI_ADAPTER_DESC ad = {};
+  GetAdapterDesc(&ad);
+
+  if (m_d3dFeatureLevel >= D3D_FEATURE_LEVEL_12_0 && ad.VendorId == 0x10DE)
```

Comment:
please add a comment why this is needed?

Reply:
Fix Nvidia stuttering on 4K playback. Added comment in code too.

Reply:
but you disable this for all res on nvidia not only for 4K. please try to move this to DXVA.cpp before creating video context.

Reply:
The issue is not 4K specific. Is Nvidia specific.
Only that ein 4K HDR is more evident (heavy stuttering).

Moved to DXVA.cpp

Reply:
ok. if it will appears only for FL12.x my be just don't use this FL?

Reply:
Negative, it's not related to FL12. Some users of v19 master branch have this issue too and users of other HDR fork that not uses FL12.

Reply:
but if it isn't related to 12.x why you disable shared textured for 12.x only?
on other side when you disable shared textures you raises back the old issue with synchronization between UI thread and decoder thread which causes shuttering, because there is no time to decoder's work.

Reply:
> why you disable shared textured for 12.x only?

Not disable it for FL12, disabled in Windows 10 because I tested it and works better (for HDR and 1080p) without side effect.

Since I not tested on Windows 7, 8 leave as is for these systems.

Reply:
there are another feature levels on Win10 and it depends on a driver not on windows version. driver may select FL11 on windows 10 also.
and iirc the issue with sync appears on latest nvidia cards. in other words you will rise the issue back definitely.

Reply:
Sorry, I'm just trying to fix a problem that was already present in v19. Really the root cause has nothing to do with HDR or this fork as I have tried to explain.

Also new code (moved at DXVA.cpp following your advice) already takes into account Windows 10 and not FL.

Reply:
Please don't take me so aggressively. I'm also want to help and fix the issue, but your solution hides the issue **of a driver** and brings back another issue of Kodi. Do you think it's a good idea? I think definitely not. Please read the old thread https://forum.kodi.tv/showthread.php?tid=339004 I don’t remember exactly how this was fixed quickly, but  it was decided a bit time after to separate the device contexts (I meant D3DDeviceContext) for the UI and for the decoder, which solved the issue of blocking the decoder by the UI thread. Before this the UI thread sleeps all time just because it has nothing to do and it sleeps in anticipation of vsync and locks the device context. Disabling sharing textures will brings back the single context for the UI and decoder and as side effect will bring back the issue with blocking decoder.

Reply:
Re-tested today with Nvidia GTX 1650, Windows build 2004 and NV driver 446.14 and results this workaround is no longer necessary. No stuttering at all.

So this code fragment is removed now.

Sorry by all noise caused.

If some kodi user still experiment stuttering can use advancedsettings or simply update Nvidia driver and Windows build.

Reply:
One forum user with GTX 1060 also reports that works fine without disabling shared textures (same code that master branch).

https://forum.kodi.tv/showthread.php?tid=349861&pid=2956619#pid2956619

Reply:
ok then

Reply:
One user with RTX 2080 Ti has reported is still experimenting stuttering on Windows 10 2004

I created new advancedsettings to has possibility of enable this workaround again.

https://github.com/thexai/xbmc/commit/3d69f228bc148b4204189722bba8f8975f90c375